lib/archethic/shared_secrets.ex

```diff
@@ -212,4 +218,83 @@ defmodule Archethic.SharedSecrets do
   def origin_family_from_public_key(<<_curve_id::8, origin_id::8, _public_key::binary>>) do
     get_origin_family_from_origin_id(origin_id)
   end
+
+  @doc """
+  Determines syncronisation of the protocol chain "node shared secrets"
+
+  How? ====>
+    mathematical || manual verification
+
+    When already in sync will return true
+
+     CurrentTime: (15thJan 23:58)
+       expected last_scheduling_date 15thJan 23:50
+       observed validation timestamp 15thJan 23:50
+
+     validation_timestamp >= last_scheduling_date
+     true
+
+    When not in sync will return false
+
+     CurrentTime: (16thJan 23:58)
+       expected last_scheduling_date  16thJan 23:50
+       observed validation timestamp  15thJan 23:50
+     validation_timestamp >= last_scheduling_date
+     false
+
+     Validation time is same as time at which tx was created.
+
+   manual:
+
+   Quorum last address from the network and compare locally
+
+  """
+  @spec is_synchronized?(:node_shared_secrets) ::
+          {:ok, :node_in_sync} | {:error, binary() | :node_out_of_sync}
+  def is_synchronized?(:node_shared_secrets) do
+    case validate_scheduling_time() do
+      true ->
+        {:ok, :node_in_sync}
+
+      false ->
+        validate_last_address()
+    end
+  end
+
+  @spec validate_scheduling_time() :: boolean()
+  def validate_scheduling_time() do
+    with genesis_address when is_binary(genesis_address) <- genesis_address(:node_shared_secrets),
+         last_scheduling_date <- get_last_scheduling_date(DateTime.utc_now()),
+         {:ok, %Transaction{validation_stamp: %{__struct__: _, timestamp: validation_timestamp}}} <-
+           TransactionChain.get_last_transaction(genesis_address, validation_stamp: [:timestamp]) do
+      validation_timestamp >= last_scheduling_date
+    else
+      _ -> false
+    end
+  end
+
+  @spec validate_last_address() :: {:ok, :node_in_sync} | {:error, binary() | :node_out_of_sync}
+  def validate_last_address() do
+    case genesis_address(:node_shared_secrets) do
+      nil ->
```

Comment:
It's not necessary to verify if the genesis_address is not nil as this function will never be called if the self repair failed which assign the genesis addresses.